File: xadrez.java
enum Cor {
    BRANCO,
    PRETO
}

enum TipoPeca {
    REI,
    RAINHA,
    TORRE,
    BISPO,
    CAVALO,
    PEAO
}

public class Peca {
    private TipoPeca tipo;
    private Cor cor;
    private int pos_x;
    private int pos_y;
    public void mover(int nova_pos_x, int nova_pos_y) {}
    public void capturar() {}
    public boolean verificarMovimento(int nova_pos_x, int nova_pos_y) { return false; }
    public void info(TipoPeca tipo, Cor cor, int pos_x, int pos_y) {}
}

File: xadrez.hs
data Cor = Branco | Preto

data TipoPeca = Rei | Rainha | Torre | Bispo | Cavalo | Peao

data Peca = Peca {
    tipo :: TipoPeca,
    cor :: Cor,
    pos_x :: Int,
    pos_y :: Int
}

mover :: Peca -> Int -> Int -> Peca
mover peca nova_pos_x nova_pos_y = undefined

capturar :: Peca -> Peca
capturar peca = undefined

verificarMovimento :: Peca -> Int -> Int -> Bool
verificarMovimento peca nova_pos_x nova_pos_y = undefined

info :: Peca -> (TipoPeca, Cor, Int, Int)
info peca = undefined
